firmware/quadruna/BMS/test/test_imd.h

```diff
@@ -22,14 +22,14 @@ class ImdTest : public testing::Test
     }
 };
 
-static void test_imd_setImdCondition(ImdConditionName condition_name)
+static void test_imd_setImdCondition(const ImdConditionName condition_name)
 {
-    const float mapping[NUM_OF_IMD_CONDITIONS] = {
-        [IMD_CONDITION_SHORT_CIRCUIT] = 0.0f,          [IMD_CONDITION_NORMAL] = 10.0f,
-        [IMD_CONDITION_UNDERVOLTAGE_DETECTED] = 20.0f, [IMD_CONDITION_SST] = 30.0f,
-        [IMD_CONDITION_DEVICE_ERROR] = 40.0f,          [IMD_CONDITION_GROUND_FAULT] = 50.0f,
+    const std::map<ImdConditionName, float> mapping{
+        { IMD_CONDITION_SHORT_CIRCUIT, 0.0f },          { IMD_CONDITION_NORMAL, 10.0f },
+        { IMD_CONDITION_UNDERVOLTAGE_DETECTED, 20.0f }, { IMD_CONDITION_SST, 30.0f },
+        { IMD_CONDITION_DEVICE_ERROR, 40.0f },          { IMD_CONDITION_GROUND_FAULT, 50.0f }
     };
```

Comment:
Again - just making sure this isn't a leak from another PR/if you maybe want to split it to another PR... I understand the appeal of just getting this in so I'm cool with that, just lmk

Reply:
was breaking on my computer

Reply:
Interesting - cool cool!